staging/src/k8s.io/cloud-provider/credentialconfig/registry.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentialconfig
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Code taken from /pkg/credentialprovider/config.go
+
+// RegistryConfig represents the config file used by the docker CLI.
+// This config that represents the credentials that should be used
+// when pulling images from specific image repositories.
+type RegistryConfig map[string]RegistryConfigEntry
+
+// RegistryConfigEntry wraps a docker config as a entry
+type RegistryConfigEntry struct {
+	Username string
+	Password string
+	Email    string
+}
+
+// registryConfigEntryWithAuth is used solely for deserializing the Auth field
+// into a dockerConfigEntry during JSON deserialization.
+type registryConfigEntryWithAuth struct {
```

Comment:
Ideally `k8s.io/cloud-provider` shouldn't include this -- if the GCP implementation needs it, can we add this to a local package there?

Reply:
GCP does need the serialization/deserialization parts, yeah. As far as I can tell no other cloud providers need it, so I'll go ahead and move it to a GCP package - thanks for pointing this out!